### ISSUE TITLE
With Astroid 3.3.10 and .pyi files, Pylint gives false duplicate code errors

### ISSUE THREAD

**Issue by avylove** [PyCQA/pylint#10418]
### Bug description

In the [blessed repo](https://github.com/jquast/blessed), we started getting false positives for duplicate code where it would have an error under a file for every file in the package. The file the bug occurs under isn't always the same. Only occurs with Astroid 3.3.10. Last successful run used 3.3.9. I tried setting jobs to 1, but it didn't make a difference.

I wasn't sure if this was an Astroid error or Pylint error, but figured I should report here since the error manifests in Pylint.

### Configuration

```ini
[SIMILARITIES]
ignore-imports=yes
min-similarity-lines=8
```

### Command used

``pylint blessed`` called from tox
tox command ``tox -e pylint``

### Pylint output

Sample below or view [CI log.txt](https://github.com/user-attachments/files/20660263/CI.log.txt).

```python
************* Module blessed._compat
blessed/_compat.py:1: [R0801(duplicate-code), ] Similar lines in 2 files
==blessed.colorspace:[17:973]
==blessed.colorspace:[17:973]
...
blessed/_compat.py:1: [R0801(duplicate-code), ] Similar lines in 2 files
==blessed.terminal:[42:1530]
==blessed.terminal:[42:1530]
...
blessed/_compat.py:1: [R0801(duplicate-code), ] Similar lines in 2 files
==blessed.keyboard:[14:476]
==blessed.keyboard:[14:476]
...
```

### Expected behavior

Same behavior as with Astroid 3.3.9

### Pylint version

```shell
Astroid Version: 3.3.10
Pylint Verision: 3.3.7
CPython: 3.14.4
```

### OS / Environment

Seen on Fedora 42 and ubuntu-latest in github CI

### Additional dependencies

```python

```

**Comment by avylove** [comment]
It seems the issue is with the handling of ``.pyi`` files. As I delete them, the files are no longer flagged as duplicate code. This library is still backward compatible to Python 2.7, so type definitions are external. I'm guessing this is not being tested?

**Comment by Pierre-Sassoulas** [comment]
Indeed we need better pyi support in astroid.
